tests/test_acasclient.py

```diff
@@ -222,22 +223,76 @@ def create_thing_with_blob_value(code):
     return ls_thing, file_name, bytes_array
 
 
-class TestAcasclient(unittest.TestCase):
-    """Tests for `acasclient` package."""
+def delete_backdoor_user(username):
+    """ Deletes a backdoor user created for testing purposes """
+    r = requests.delete(ACAS_NODEAPI_BASE_URL + "/api/systemTest/deleteTestUser/" + username)
+    r.raise_for_status()
+
+
+def create_backdoor_user(username, password, acas_user=True, acas_admin=False, creg_user=False, creg_admin=False, project_names=None):
+    """ Creates a backdoor user for testing purposes """
+    body = {
+        "username": username,
+        "password": password,
+        "acasUser": acas_user,
+        "acasAdmin": acas_admin,
+        "cmpdregUser": creg_user,
+        "cmpdregAdmin": creg_admin,
+        "projectNames": project_names or []
+    }
+    r = requests.post(ACAS_NODEAPI_BASE_URL + "/api/systemTest/getOrCreateTestUser", json=body)
+    r.raise_for_status()
+    return r.json()
+
 
+def get_or_create_global_project():
+    """ Creates a global project for testing purposes """
+    r = requests.get(ACAS_NODEAPI_BASE_URL + "/api/systemTest/getOrCreateGlobalProject")
+    r.raise_for_status()
+    output = r.json()
+    return output["messages"]
+
+class BaseAcasClientTest(unittest.TestCase):
+    """ Base class for ACAS Client tests """
     def setUp(self):
         """Set up test fixtures, if any."""
         creds = acasclient.get_default_credentials()
-        self.client = acasclient.client(creds)
+        # Ensure Global project is there
+        global_project = get_or_create_global_project()
+        self.global_project_code = global_project["code"]
+        self.test_usernames = []
+        try:
+            self.client = acasclient.client(creds)
+        except RuntimeError:
+            # Create the default user if it doesn't exist
+            if creds.get('username'):
+                self.test_usernames.append(creds.get('username'))
+                create_backdoor_user(creds.get('username'), creds.get('password'), acas_user=True, acas_admin=True, creg_user=True, creg_admin=True)
+            # Login again
+            self.client = acasclient.client(creds)
         self.tempdir = tempfile.mkdtemp()
         # Set TestCase - maxDiff to None to allow for a full diff output when comparing large dictionaries
         self.maxDiff = None
 
     def tearDown(self):
         """Tear down test fixtures, if any."""
         shutil.rmtree(self.tempdir)
+        if self.test_usernames:
+            for username in self.test_usernames:
+                delete_backdoor_user(username)
         self.client.close()
```

Comment:
I believe `self.client.close()` will not be executed if an exception was raised in one of the `delete_backdoor_user` calls. Is this expected?

Reply:
Good point, wrapping this in a try/finally block